### PR TITLE
fix(barbican): validate store config and fix SecretExists error message

### PR DESCRIPTION
**Pull request by alekc**
## Problem Statement

Three small correctness issues in the Barbican provider, found during a documentation audit (#6531):

`ValidateStore` is a no-op, so a `SecretStore` that is missing `authURL` or its auth credentials passes admission and only fails later at client init during reconcile, which is a confusing late failure:

https://github.com/external-secrets/external-secrets/blob/67a603db02ef92ea0eee6b091927f2727cc54eaa/providers/v1/barbican/provider.go#L51-L56

Because `ValidateStore` does nothing, the "store validation" column for Barbican in the provider support matrix (`docs/introduction/stability-support.md`) is not actually backed by any validation.

The `SecretExists` error message is garbled:

https://github.com/external-secrets/external-secrets/blob/67a603db02ef92ea0eee6b091927f2727cc54eaa/providers/v1/barbican/client.go#L132-L135

## Related Issue

No tracking issue; follow-up to the Barbican documentation audit (#6531).

## Proposed Changes

- Implement `ValidateStore`: require `authURL`, require `auth.username` to have either a `value` or a `secretRef`, and require `auth.password.secretRef`. Misconfiguration is now caught at admission instead of at reconcile, and the "store validation" support-matrix entry is now accurate.
- Fix the `SecretExists` error message to "barbican provider does not support checking secret existence (read-only)".
- Add `ValidateStore` test cases (missing `authURL`, empty username, missing password ref, value-only username, nil provider) and update the `SecretExists` assertion.

No CRD or API-type changes.

## Checklist

- [x] I have read the [contribution guidelines](https://external-secrets.io/latest/contributing/process/#submitting-a-pull-request)
- [x] All commits are signed with `git commit --signoff`
- [x] My changes have reasonable test coverage (added `ValidateStore` cases)
- [x] All tests pass with `make test` (ran `go test` for the barbican module; the full suite runs in CI)
- [x] I ensured my PR is ready for review with `make reviewable` (ran `go test`, `go vet`, the pinned `golangci-lint` with 0 issues, and a standalone `go mod tidy` with no drift; the full reviewable target runs in CI)

Updated Barbican provider store validation to fail at admission by implementing `ValidateStore`: it now requires `authURL`, requires `auth.username` via either `value` or `secretRef` (rejecting empty/unspecified username), and requires `auth.password.secretRef`. When username/password use `secretRef`, the referenced selectors are validated to enforce required namespace rules (including cluster store cases where `SecretRef` namespaces are missing). Also clarified the read-only `SecretExists` error message to state that secret-existence checks aren’t supported. Tests were expanded for missing `authURL`, empty/ value-only username, missing password secretRef, cluster secretRef namespace behavior, and nil-provider handling; `SecretExists` assertion was updated accordingly.

### CODE REVIEW

File: providers/v1/barbican/provider.go [hunk 1]
```diff
@@ -52,6 +52,19 @@ func (p *Provider) ValidateStore(store esv1.GenericStore) (admission.Warnings, e
...
+		return nil, err
+	}
+	if provider.AuthURL == "" {
+		return nil, fmt.Errorf(errMissingField, errors.New("authURL is required"))
+	}
+	if provider.Auth.Username.Value == "" && provider.Auth.Username.SecretRef == nil {
+		return nil, fmt.Errorf(errMissingField, errors.New("auth.username requires either value or secretRef"))
+	}
+	if provider.Auth.Password.SecretRef == nil {
+		return nil, fmt.Errorf(errMissingField, errors.New("auth.password.secretRef is required"))
```
[the remark this answers is not in the document]
Comment (alekc) [reply providers/v1/barbican/provider.go:72]: Good call, done in 2b4af684e. Both auth secretRefs now go through `esutils.ValidateSecretSelector`, so the namespace rules are enforced in `ValidateStore` (a ClusterSecretStore without a `secretRef.namespace` now fails admission). Added cluster-store test cases.

**Review by Skarlso — APPROVED** [review]
